decidim-proposals/app/commands/decidim/proposals/update_proposal.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # A command with all the business logic when a user updates a proposal.
+    class UpdateProposal < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form         - A form object with the params.
+      # current_user - The current user.
+      # proposal - the proposal to update.
+      def initialize(form, current_user, proposal)
+        @form = form
+        @current_user = current_user
+        @proposal = proposal
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the proposal.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
```

Comment:
It's strange to me that the `time_limit` isn't checked here, but is instead implicit as a `cancan` permission. It's too brittle of a security model.

Maybe we should move the time limit check into `proposal.editable?` and then check it here. The whole reason for commands to exist is to serve as `gateways` for actions, so all the relevant logic involved is easily readable, instead of spread amongst model validations, callbacks and controllers.

Also, I don't think we're preventing proposals from being edited when they're accepted / rejected / in evaluation. We should definitely do that, as the time limit could be potentially very high.

Reply:
+1 to the `proposal.editable?` part. 

> Also, I don't think we're preventing proposals from being edited when they're accepted / rejected / in evaluation. We should definitely do that, as the time limit could be potentially very high.

This is done in the `cancancan` ability, but can be moved into the method anyway.

Reply:
Done!